pkg/dbadmin/mysqladmin/admin.go

```diff
@@ -265,12 +267,124 @@ func (mdba *MySQLDbAdmin) GetNextIDs(tableNames []dbadmin.TableName) (map[dbadmi
 // SelectFloat implements DbAdmin
 func (mdba *MySQLDbAdmin) SelectFloat(selectQuery string) (result float64, _ error) {
 	if err := mdba.handle.Get(&result, selectQuery); err != nil {
-		return result, fmt.Errorf("Unable to select float value from the database: %w", err)
+		return result, wrap(err)
 	}
 
 	return result, nil
 }
 
+// IsBlockingIndexCreation implements DbAdmin
+func (mdba *MySQLDbAdmin) IsBlockingIndexCreation(tableName dbadmin.TableName, indexType dbadmin.IndexType, columns ...string) (bool, error) {
+	return false, fmt.Errorf("Not implemented")
+}
+
+// ConstraintWillFail implements DbAdmin
+func (mdba *MySQLDbAdmin) ConstraintWillFail(tableName dbadmin.TableName, constraintType dbadmin.ConstraintType, columns ...string) (bool, error) {
+	switch constraintType {
+	case dbadmin.NotNullConstraint:
+		// Find out if the table already has a not null constraint on all of the named columns
+		query, args, err := sqlx.In("SHOW COLUMNS FROM `?` WHERE Field IN (?);", tableName, columns)
+		if err != nil {
+			return true, fmt.Errorf("Unable to prepare query to read table definition: %w", err)
+		}
+		query = mdba.handle.Rebind(query)
+
+		var fieldDefinitions []struct {
+			Field string
+			Type  string
+			Null  string
+		}
+		if err := mdba.handle.Select(&fieldDefinitions, query, args...); err != nil {
+			return true, wrap(err)
+		}
+
+		fieldNullableMap := make(map[string]string, len(fieldDefinitions))
+		for _, fieldDefinition := range fieldDefinitions {
+			fieldNullableMap[fieldDefinition.Field] = fieldDefinition.Null
+		}
+
+		for _, columnToCheck := range columns {
+			definition, ok := fieldNullableMap[columnToCheck]
+			if !ok {
+				// Will definitely fail because the column doesn't exist
+				return true, nil
```

Comment:
I don't think so, if we just return `true` here with no error it will annotate the `ManagedDatabase` object that `Attempting to convert a column (%s) to NOT NULL on a table (%s) when null data is present`. That's ideal, right?